### PR TITLE
[execution] Output all events not just reconfig events

### PR DESCRIPTION
**Pull request by gregnazario**
## Motivation

Reconfig events were the only ones coming out of execution, let's make it so we can access other events, and just filter later

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

Run the current tests, as no functionality really should have changed

### CODE REVIEW

**Review by zekun000 — COMMENTED** [review]
this is much more expensive though since every txns emit events. I was thinking about some subscriptions initially but in the end we hard coded the reconfig one.

File: state-sync/src/executor_proxy.rs [hunk 1]
```diff
@@ -13,6 +13,7 @@ use diem_types::{
     contract_event::ContractEvent,
     ledger_info::LedgerInfoWithSignatures,
     move_resource::MoveStorage,
+    on_chain_config,
```
Comment (lightmark) [inline state-sync/src/executor_proxy.rs:16]: remove?

**Review by lightmark — APPROVED** [review]
LGTM

**Comment by davidiw** [comment]
Let's get some perf numbers. I suspect it may benefit us to have an explicit subscription list since this can become more expensive just by adding new smart contracts that emit more and more events.

**Comment by davidiw** [comment]
all up, gas price 1 : 959 TPS, 4741 ms latency, 5400 ms p99 latency, backup: -1 Bps, no expired txns, time spent 269 secs
all up : 1065 TPS, 4264 ms latency, 4850 ms p99 latency, backup: -1 Bps, no expired txns, time spent 250 secs
10% down : 1067 TPS, 3823 ms latency, 4800 ms p99 latency, backup: -1 Bps, no expired txns, time spent 406 secs
3 Region Simulation : 726 TPS, 6268 ms latency, 7850 ms p99 latency, no expired txns, time spent 267 secs
fixed tps 10 : 10 TPS, 744 ms latency, 800 ms p99 latency, backup: -1 Bps, no expired txns, time spent 248 secs
State sync throughput : 2393 txn/sec, time spent 521 secs

**Comment by gregnazario** [comment]
> wow, this is the current state of testnet... so it seems like this PR does not add a lot or really any observable overheads :O
> all up, gas price 1 : 959 TPS, 4741 ms latency, 5400 ms p99 latency, backup: -1 Bps, no expired txns, time spent 269 secs
> all up : 1065 TPS, 4264 ms latency, 4850 ms p99 latency, backup: -1 Bps, no expired txns, time spent 250 secs
> 10% down : 1067 TPS, 3823 ms latency, 4800 ms p99 latency, backup: -1 Bps, no expired txns, time spent 406 secs
> 3 Region Simulation : 726 TPS, 6268 ms latency, 7850 ms p99 latency, no expired txns, time spent 267 secs
> fixed tps 10 : 10 TPS, 744 ms latency, 800 ms p99 latency, backup: -1 Bps, no expired txns, time spent 248 secs
> State sync throughput : 2393 txn/sec, time spent 521 secs

The performance shouldn't be all that different as we're already getting the events.  We just are filtering them out at a later time.  In theory this adds some space and mild time complexity about constantly appending more events, but since they're already coming from the executor this shouldn't really cause an impact.

Now, once we start doing something with these events, that will be more clear.

**Review by zekun000 — DISMISSED** [review]
ah I was worried about the serde cost with DEC and now realized this is only about ChunkExecutor which is just in memory.

File: state-sync/src/executor_proxy.rs (hunk 1 above)
Comment (gregnazario) [reply state-sync/src/executor_proxy.rs:16]: It's used for `on_chain_config::new_epoch_event_key()`

**Pull request merged**